File: app/src/main/java/com/example/beretta/owoxgallary/arch/BaseArchViewModel.java
package com.example.beretta.owoxgallary.arch;

import android.arch.lifecycle.ViewModel;
import android.support.annotation.Nullable;

/**
 * Created by beretta on 12.10.2017.
 */

public class BaseArchViewModel<V extends ArchViewContract> extends ViewModel implements ArchViewModelContract<V>{

    @Nullable
    private V mView;
    @Override
    public void attachView(V view) {
        mView = view;
    }

    @Override
    public void detachView() {
        mView = null;
    }
}

File: app/src/main/java/com/example/beretta/owoxgallary/arch/viewmodel/BaseArchViewModel.kt
package com.example.beretta.owoxgallary.arch.viewmodel

import androidx.lifecycle.ViewModel
import com.example.beretta.owoxgallary.arch.view.ArchViewContract


open class BaseArchViewModel<V : ArchViewContract> : ArchViewModelContract<V>, ViewModel(){

    protected var view: V? = null

    override fun attachView(view: V) {
        this.view = view
    }

    override fun detachView() {
        view = null
    }
}